plasma_framework/test/src/exits/payment/PaymentInFlightExitable.test.js

```diff
@@ -457,5 +457,220 @@ contract('PaymentInFlightExitable', ([_, alice, bob, carol]) => {
+                    MIN_EXIT_PERIOD, DUMMY_INITIAL_IMMUNE_VAULTS_NUM, INITIAL_IMMUNE_EXIT_GAME_NUM,
+                );
+                this.exitGame = await PaymentInFlightExitable.new(this.framework.address);
+
+                const { args, argsDecoded, inputTxsBlockRoot } = buildValidIfeStartArgs(
+                    AMOUNT, [alice, bob, carol], BLOCK_NUMBER,
+                );
+                this.args = args;
+                this.argsDecoded = argsDecoded;
+                await this.framework.setBlock(BLOCK_NUMBER, inputTxsBlockRoot, 0);
+
+                const conditionTrue = await PaymentSpendingConditionTrue.new();
+
+                await this.exitGame.registerSpendingCondition(
+                    OUTPUT_TYPE_ZERO, IFE_TX_TYPE, conditionTrue.address,
+                );
+
+                await this.exitGame.startInFlightExit(
+                    this.args,
+                    { from: alice, value: IN_FLIGHT_EXIT_BOND },
+                );
+
+                const {
+                    args: cArgs, block, decodedCompetingTx,
+                } = createValidNoncanonicalChallengeArgs(this.argsDecoded.inFlightTx);
+
+                this.challengeArgs = cArgs;
+                this.competingTx = decodedCompetingTx;
+                this.competingTxBlock = block;
+            });
+
+            it('should successfully challenge ife', async () => {
+                await this.framework.setBlock(
+                    this.competingTxBlock.blockNum,
+                    this.competingTxBlock.blockHash,
+                    this.competingTxBlock.blockTimestamp,
+                );
+
+                const tx = await this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice });
+
+                expectEvent.inLogs(tx.logs, 'InFlightExitChallenged', {
+                    challenger: alice,
+                    txHash: web3.utils.sha3(this.args.inFlightTx),
+                    challengeTxPosition: new BN(this.challengeArgs.competingTxPos),
+                });
+            });
+
+            it('fails when competing tx is the same as in-flight one', async () => {
+                this.challengeArgs.competingTx = this.challengeArgs.inFlightTx;
+
+                await expectRevert(
+                    this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice }),
+                    'The competitor transaction is the same as transaction in-flight',
+                );
+            });
+
+            it('fails when first phase is over', async () => {
+                await time.increase((MIN_EXIT_PERIOD / 2) + 1);
+
+                await expectRevert(
+                    this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice }),
+                    'Canonicity challege phase for this exit has ended',
+                );
+            });
+
+            it('fails when competing tx is not included', async () => {
```

Comment:
I might be a bit more explicit since it is possible to challenge with inflight competing transaction.
```
fails when competing tx is not included in the given position
```